### DISCUSSION TITLE
Liquidsoap metadata override

### DISCUSSION THREAD

**Discussion by d-trax** [AzuraCast/AzuraCast#7144] [General]
Hey guys, I am struggling here with setting custom metadata (instead of showing the Artist and Title to the stream), to show the listeners the name of the radio station. I've done this in past Liquidsoap versions (1.4.4), but it seems that the script doesn't work with Liquidsoap 2.x. 
The script I was using:

```
# Metadata override
def override_metadata(m) =
 [("title","Text replacing song title"),("artist","Text replacing song artist")]
end
radio = map_metadata(override_metadata, radio)
```
I believe that there is something changed in the language, and that's why the script doesn't work anymore. Does somebody know a script, that should work? Thanks in advance!

**Comment by gAlleb** [comment]
Hi, @d-trax. 

You're right. The best option would be something like:

```ruby
def override_metadata(m) =
  [("artist", "Your"), ("title", "Text")]
end

radio = metadata.map(update=false, override_metadata, radio)
```

That's it) change `radio = map_metadata(override_metadata, radio)` to `radio = metadata.map(update=false, override_metadata, radio)`

**Reply by d-trax** [reply]
Thanks bro, that worked!

**Comment by d-trax** [comment]
Got another question. Is there a chance to override the metadata only to Icecast/listeners, because I want to be able to see on my main radio page, what's playing atm. I believe that on older Icecast versions that can be done with 'icy_metadata', but since Icecast2, that's not an option anymore.

**Reply by gAlleb** [reply]
The above function won't affect you public radio page (e.g. `https://radio.com/public/radio`).  What exactly do you mean?

**Reply by d-trax** [reply]
I mean, when using the scrip you provided (thanks for that), the public player shows a text which I input, but in my main radio control panel I see the exact text, not the song's artist and title. I believe if the script manipulates only the Icecast's metadata, the song's title and artist would appear in the radio's control panel. Attaching a screenshot.
![Screen Shot 2024-05-25 at 2 00 31 PM](https://github.com/AzuraCast/AzuraCast/assets/31935215/f0a77193-e9da-45e5-bbbc-4b09fabfced0)
Long story short - I want my listeners to see the text (not the song artist and title) and in my control panel I want to see the actual song artist and title.

**Reply by gAlleb** [reply]
I've just tested on my instance.

```ruby
def override_metadata(m) =
  [("artist", "Your"), ("title", "Text")]
end

radio = metadata.map(update=false, override_metadata, radio)
```
This goes to the box before `# Local Broadcasts`. Don't forget `update=false`.

And I see on public player `real` song and title. In control panel I see  `real` song and title. But Icecast2 sends `Your - text`

<img width="500" alt="Снимок экрана 2024-05-25 в 14 36 32" src="https://github.com/AzuraCast/AzuraCast/assets/144447266/afc2f5f4-0d50-4ade-9de8-c140caec2b29">
<img width="500" alt="Снимок экрана 2024-05-25 в 14 36 32" src="https://github.com/AzuraCast/AzuraCast/assets/144447266/8ffd493d-f80e-4ba6-a41a-35ffa30ca822">

**Reply by gAlleb** [reply]
If this doesn't work for you (I can't see reason why it shouldn't). 

I can only think of a non-conventional method)

Say, you have 2 icecast mount points - `radio.mp3` and `radio.aac`

1. Go to `Station` -> `Broadcasting` -> `Edit Liquidsoap Configuration` and scroll to the end. You will see something like this:

```ruby
# Local Broadcasts
output.icecast(%mp3(samplerate=44100, stereo=true, bitrate=128), id="local_1", host = "127.0.0.1", port = 8000, password = "", mount = "/radio.mp3", name = "Name", description = "Hard Rock fans never die! They just multiply!", genre = "", url = "", public = false, encoding = "UTF-8", radio)

output.icecast(%fdkaac(........), id="local_1", host = "127.0.0.1", port = 8000, password = "", mount = "/radio.aac", name = "Name", description = "Hard Rock fans never die! They just multiply!", genre = "", url = "", public = false, encoding = "UTF-8", radio)
```
Copy it somewhere.

2. Now go to  `Station` -> `Broadcasting` -> `Mountponts`. You will see your mountpoints `radio.mp3` and `radio.aac`. Click `Edit` -> `AutoDJ` and uncheck `Enable AutoDJ` for both of them. This way we disable Liquidsoap for this mountpoints.

3. Go to `Station` -> `Broadcasting` -> `Edit Liquidsoap Configuration` and scroll to the end. `output.icecast` should be gone.
Before `# Local Broadcasts` there is a box. Input this

```ruby
icecast_radio = radio 

def override_metadata(m) =
  [("artist", "Your"), ("title", "Text")]
end

icecast_radio = metadata.map(update=false, override_metadata, icecast_radio)

# next insert manually your icecast outputs (that you've copied before) but substitute `radio` for `icecast_radio` var

output.icecast(%mp3(samplerate=44100, stereo=true, bitrate=128), id="local_1", host = "127.0.0.1", port = 8000, password = "", mount = "/radio.mp3", name = "Name", description = "Hard Rock fans never die! They just multiply!", genre = "", url = "", public = false, encoding = "UTF-8", icecast_radio)

output.icecast(%fdkaac(........), id="local_1", host = "127.0.0.1", port = 8000, password = "", mount = "/radio.aac", name = "Name", description = "Hard Rock fans never die! They just multiply!", genre = "", url = "", public = false, encoding = "UTF-8", icecast_radio)
```

Restart station.

But again 

```
def override_metadata(m) =
  [("artist", "Your"), ("title", "Text")]
end

radio = metadata.map(update=false, override_metadata, radio)
```
Is totally enough in the first place.

**Reply by d-trax** [reply]
Tried and the script works only in the box after `add_skip_command(radio)`
If I add it after # Local Broadcasts, it doesn't work at all.

Btw I am using old instance of Azuracast (0.17.4) with Ansible. What version are you using?

**Reply by gAlleb** [reply]
Not after # Local Broadcasts - Before

<img width="500" alt="Снимок экрана 2024-05-25 в 14 36 32" src="https://github.com/AzuraCast/AzuraCast/assets/144447266/012d6469-5981-4f15-b7df-9e2b48032515">

> What version are you using?

Latest RR. `0.17.4` is very old. You should consider upgrading.

I don't know whether you have an input box before "# Local Broadcasts". If you don't  and have only icecast outputs then use this instruction https://github.com/AzuraCast/AzuraCast/discussions/7144#discussioncomment-9555120 and paste the contents into the box after "# Local Broadcasts"
